docs/acceptance-criteria/uacs-tbtc-v2.adoc

```diff
@@ -0,0 +1,291 @@
+  challenge period, then: +
+  - challenge transaction is reverted.
+
+| Wallet lifecycle - wallet creation - DKG approval
+| Anyone can request unlocking of the sortition pool and marking of the DKG
+  result as approved (but not all requests will be processed positively).
+
+| Wallet lifecycle - wallet creation - DKG approval
+| Upon processing the transaction unlocking the sortition pool and marking the
+  DKG result as approved, the DKG contract should determine the validity of the
+  transaction.
+
+| Wallet lifecycle - wallet creation - DKG approval
+| When determining the validity of the tx unlocking the sortition pool and
+  approving the DKG result, the following aspects should be taken into
+  account: +
+  - is the sortition pool locked? +
+  - has the challenge period already passed? +
+  - is the sender eligible to approve the DKG result? +
+  Only if all the above conditions have been met, the transaction should be
+  considered valid.
+
+// This UAC (and the name of the param) is based on the assumption that the
+// result submission conditions will be the same as for Random Beacon (at first
+// one operator can submit, after `resultSubmissionEligibilityDelay` another is
+// added to the list of operators eligible to publish the result, and so on...).
+// This is what initial implementation of a DKG fo ECDSA looks like, but there
+// are ongoing discussions whether this is the best approach. We may end up not
+// having `resultSubmissionEligibilityDelay` param.
+| Wallet lifecycle - wallet creation - DKG approval
+| At the beginning (right after DKG result is submitted on chain) only the
+  result submitter should be eligible to accept the DKG result and get the
+  reimbursement.
+  After a fixed amount of time (`resultSubmissionEligibilityDelay`) everybody
+  should become eligible to approve the DKG result.
+
+| Wallet lifecycle - wallet creation - DKG approval
+| If the transaction unlocking the sortition pool and marking the DKG result as
+  approved is deemed valid, then: +
+  - the sortition pool should get unlocked +
+  - the DKG result should be approved and a new wallet should be created in the
+  wallets contracts based on a candidate group (minus inactive/disqualified
+  members) +
+  - the DKG result accepter should receive reimbursement for the gas spent on
+  the DKG submit and DKG approve transactions, paid from the DAO-funded ETH pool
+  to the notifier’s address (in the same tx); there should be a limit on the
+  refunded gas price (`max_gas_refund_price`) (if the amount of gwei paid for
+  gas in the DKG result submission is bigger than this value, the submitter will
+  be reimbursed `max_gas_refund_price x gas_used`) +
+  - operators marked as inactive/disqualified during DKG protocol execution
+  should be punished by being excluded from earning rewards for a fixed amount
+  of time (`sortitionPoolRewardsBanDuration`) (in the same tx).
```

Comment:
Is using a reward ban something that we'll go with? There are ongoing discussions on how to handle this so that the cost of banning somebody would not be higher than savings made by banning him. With the current idea we also need to figure out how to correctly prolongate a ban on an operator which already is banned.